src/lib/io/include/sol-ipm.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sol-types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @brief These routines are used for Inter Processor Messaging
+ * on Soletta.
+ */
+
+/**
+ * @defgroup IPM IPM
+ * @ingroup IO
+ *
+ * @brief Inter Processor Messaging API for Soletta.
+ *
+ * Some platforms may have more than one core available to use. This API
+ * provides a way to send data from one to another, assuming shared
+ * memory among cores and a mechanism to send messages from one core to
+ * another, like mailbox.
+ *
+ * @{
+ */
+
+/**
+ * @brief Set receiver to IPM messages.
+ *
+ * Receives messages sent with id @a id on callback @a receive_cb. Message
+ * content is on callback parameter @a message.
+ *
+ * @param id id of messages to listen to. Only messages sent with @a id
+ * will be sent to this receiver.
+ * @param receive_cb callback to be called with message sent with @a id.
+ * If @c NULL, deletes a previously set callback.
+ * @param data user defined data that will be sent to callback @a cb
+ *
+ * @return 0 if receiver was set (or unset if @a receive_cb is @c NULL).
+ * Negative number otherwise.
+ *
+ * @note message received is a @c sol_blob. Only when all references to this
+ * blob are gone sender will receive its consumed confirmation.
+ *
+ * @attention Soletta reserves id 0 - do not use it.
```

Comment:
On 09-05-2016 16:04, Gustavo Sverzut Barbieri wrote:
> In src/lib/io/include/sol-ipm.h
> <https://github.com/solettaproject/soletta/pull/1996#discussion_r62554382>:
>
>> + * Receives messages sent with id @a id on callback @a receive_cb. Message
>> + * content is on callback parameter @a message.
>> + *
>> + * @param id id of messages to listen to. Only messages sent with @a id
>> + * will be sent to this receiver.
>> + * @param receive_cb callback to be called with message sent with @a id.
>> + * If @c NULL, deletes a previously set callback.
>> + * @param data user defined data that will be sent to callback @a cb
>> + *
>> + * @return 0 if receiver was set (or unset if @a receive_cb is @c NULL).
>> + * Negative number otherwise.
>> + *
>> + * @note message received is a @c sol_blob. Only when all references to this
>> + * blob are gone sender will receive its consumed confirmation.
>> + *
>> + * @attention Soletta reserves id 0 - do not use it.
>
> this needs to relate to the build-time option. You can mention by
> default it's 0. I guess we need a getter for that value as well :-/

No, channel id is different from message id. Soletta uses always 0 as 
message id, not channel id. Build time options are for channels id.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1996/files/64ac7ea3f07d3a26a3d264869f28dc376ba64cc9#r62554382>
>